lib/controllers/hosted-skill-controller/helper.js

```diff
@@ -81,3 +95,74 @@ function handleSkillStatus(response, skillId, callback) {
         callback(`${errMessage}${errReason}Infrastructure provision for the hosted skill failed. Please try again.`);
     }
 }
+
+function downloadAuthInfoScript(callback) {
+    const authInfoUrl = CONSTANTS.HOSTED_SKILL.S3_SCRIPTS.AUTH_INFO.URL;
+    const askFolderPath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER);
+    const authInfoPath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER,
+        CONSTANTS.FILE_PATH.ASK.AUTH_INFO);
+    if (!fs.existsSync(askFolderPath)) {
+        fs.mkdirSync(askFolderPath);
+    }
+    this.downloadScriptFromS3(authInfoUrl, authInfoPath, (err) => callback(err || null));
+}
+
+function downloadAskPrePushScript(callback) {
+    const askScriptUrl = CONSTANTS.HOSTED_SKILL.S3_SCRIPTS.ASK_PRE_PUSH.URL;
+    const askFolderPath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER);
+    const scriptFolderPath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.NAME);
+    const askFilePath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.NAME,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.ASK_PRE_PUSH);
+    if (!fs.existsSync(askFolderPath)) {
```

Comment:
you can just use 
https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir-sync.md